src/main/java/org/apache/ibatis/executor/resultset/DefaultResultSetHandler.java

```diff
@@ -691,29 +687,17 @@ private Object createUsingConstructor(ResultSetWrapper rsw, Class<?> resultType,
     return null;
   }
 
-  private boolean allowedConstructor(final Constructor<?> constructor, final List<String> classNames) {
+  private boolean allowedConstructorUsingTypeHandlers(final Constructor<?> constructor, final List<JdbcType> jdbcTypes) {
     final Class<?>[] parameterTypes = constructor.getParameterTypes();
-    if (typeNames(parameterTypes).equals(classNames)) return true;
-    if (parameterTypes.length != classNames.size()) return false;
+    if (parameterTypes.length != jdbcTypes.size()) return false;
     for (int i = 0; i < parameterTypes.length; i++) {
-      final Class<?> parameterType = parameterTypes[i];
-      if (parameterType.isPrimitive() && !primitiveTypes.getWrapper(parameterType).getName().equals(classNames.get(i))) {
-        return false;
-      } else if (!parameterType.isPrimitive() && !parameterType.getName().equals(classNames.get(i))) {
+      if (!typeHandlerRegistry.hasTypeHandler(parameterTypes[i], jdbcTypes.get(i))) {
         return false;
       }
     }
     return true;
   }
```

Comment:
Yea that means that multiple constructors it will choose the first that passes.
I think that's ok as it works for the majority of the cases.
As you mentioned in the previous code, choosing correct types isn't so simple.
TypeHandlers is a very powerful feature and leveraging it is definitely a positive.
It's also more consistent with how automapping are currently done.
As for the old allowedConstructor, it was more restrictive as initially my aim was just to slightly improve the existing autoconstructor mapping.

Reply:
I see. If it's intentional, I'm OK with it. :)